iothub/device/src/Transport/Mqtt/MqttTransportSettings.cs

```diff
@@ -86,48 +70,141 @@ public MqttTransportSettings(TransportType transportType)
 
+        /// <summary>
+        /// Indicates if max outbound retransmission is enforced.
+        /// The default value is <c>false</c>.
+        /// </summary>
+        /// <remarks>
+        /// This property is currently unused.
+        /// </remarks>
         public bool MaxOutboundRetransmissionEnforced { get; set; }
 
+        /// <summary>
+        /// The maximum no. of inbound messages that are read from the channel.
+        /// The default value is 50.
+        /// </summary>
         public int MaxPendingInboundMessages { get; set; }
 
+        /// <summary>
+        /// The time to wait for receiving an acknowledgment for a CONNECT packet.
+        /// The default is 10 seconds.
+        /// </summary>
         public TimeSpan ConnectArrivalTimeout { get; set; }
 
         /// <summary>
         /// Flag to specify if a subscription should persist across different sessions. The default value is false.
-        /// If set to false: the device will receive messages that were sent to it while it was disconnected.
-        /// If set to true: the device will receive only those messages that were sent to it after it successfully subscribed to the device bound message topic.
         /// </summary>
+        /// <remarks>
+        /// <para>If set to false: the device will receive messages that were sent to it while it was disconnected.</para>
+        /// <para>If set to true: the device will receive only those messages that were sent to it
+        /// after it successfully subscribed to the device bound message topic.</para>
+        /// </remarks>
         public bool CleanSession { get; set; }
 
+        /// <summary>
+        /// The interval, in seconds, that the client establishes with the service, for sending keep alive pings.
+        /// The default is 300 seconds.
+        /// </summary>
+        /// <remarks>
+        /// The client will send a ping request 4 times per keep-alive duration set.
+        /// It will wait for 30 seconds for the ping response, else mark the connection as disconnected.
+        /// </remarks>
         public int KeepAliveInSeconds { get; set; }
 
         /// <summary>
-        /// Whether the transport has a will message
+        /// Indicates whether the transport has a will message.
         /// </summary>
+        /// <remarks>
+        /// Setting a will message is a way for clients to notify other subscribed clients about ungraceful disconnects in an appropriate way.
+        /// In response to the ungraceful disconnect, the service will send the last-will message to the configured telemetry channel.
+        /// The telemetry channel can be either the default Events endpoint or a custom endpoint defined by IoT Hub routing.
+        /// For more details, refer to https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-mqtt-support#using-the-mqtt-protocol-directly-as-a-device.
+        /// </remarks>
         public bool HasWill { get; set; }
```

Comment:
It is too late to change this property to HasWillMessage ? is this publicly exposed?

Reply:
We cannot delete them. We have to either hook them up, or mark them as deprecated.
I am still waiting for confirmation from the service team on if these properties are required. This is to indicate to customer that setting this property right now has no effect.

Reply:
Yeah, `HasWill` is already public